src/Libraries/RevitNodes/Elements/ElementType.cs

```diff
@@ -0,0 +1,200 @@
+using RevitServices.Transactions;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Revit.Elements
+{
+    /// <summary>
+    /// A Revit ElementType
+    /// </summary>
+    public class ElementType : Element
+    {
+        #region Internal properties
+
+        /// <summary>
+        /// An internal reference to the ElementType
+        /// </summary>
+        internal Autodesk.Revit.DB.ElementType InternalElementType
+        {
+            get; private set;
+        }
+
+        /// <summary>
+        /// Reference to the Element
+        /// </summary>
+        public override Autodesk.Revit.DB.Element InternalElement
+        {
+            get { return InternalElementType; }
+        }
+
+        #endregion
+
+        #region Private constructors
+
+        /// <summary>
+        /// Private constructor for the Element
+        /// </summary>
+        /// <param name="elementType"></param>
+        private protected ElementType(Autodesk.Revit.DB.ElementType elementType)
+        {
+            SafeInit(() => InitElementType(elementType));
+        }
+
+        /// <summary>
+        /// Initialize a ElementType element
+        /// </summary>
+        /// <param name="elementType"></param>
+        private void InitElementType(Autodesk.Revit.DB.ElementType elementType)
+        {
+            InternalSetElementType(elementType);
+        }
+
+        #endregion
+
+        #region Private mutators
+
+        /// <summary>
+        /// Set the ElementType property, element id, and unique id
+        /// </summary>
+        /// <param name="elementType"></param>
+        private void InternalSetElementType(Autodesk.Revit.DB.ElementType elementType)
+        {
+            this.InternalElementType = elementType;
+            this.InternalElementId = elementType.Id;
+            this.InternalUniqueId = elementType.UniqueId;
+        }
+
+        #endregion
+
+        #region Public properties
+
+        /// <summary>
+        /// The name of the ElementType
+        /// </summary>
+        public new string Name
+        {
+            get { return InternalElementType.Name; }
+        }
+
+        /// <summary>
+        /// The FamilyName of the ElementType
+        /// </summary>
+        public  string FamilyName
+        {
+            get { return InternalElementType.FamilyName; }
+        }
+
+        /// <summary>
+        /// Determine if this ElementType can be deleted
+        /// </summary>
+        public bool CanBeDeleted
+        {
+            get { return InternalElementType.CanBeDeleted; }
+        }
+
+        /// <summary>
+        /// Determine if this ElementType can create a copy
+        /// </summary>
+        public bool CanBeCopied
+        {
+            get { return InternalElementType.CanBeCopied; }
+        }
+
+        /// <summary>
+        /// Determine if this ElementType can be renamed
```

Comment:
let's make all user-facing descriptions consistent, either end them with a `.` or not - dunno what the current pattern is, from what i see it's ending with a colon `.`